src/cmd/ksh93/tests/subshell.sh

```diff
@@ -726,5 +726,18 @@ check_hash_table()
 (hash cat)
 [[ $(hash) == "chmod=$(whence -p chmod)" ]] || err_exit $'changes to a subshell\'s hash table affect the parent shell'
 
+# ======
+# Variables set in functions inside of a virtual subshell should not affect the
+# outside environment. This regression test must be run in a here document.
```

Comment:
You are not actually running the regression test in a here-document; you are writing a shell script to disk using a here-document, then running it in the regular way like `ksh file.sh`. Which is fine, but that comment is a bit misleading…

You could either fix the comment, or fix the code to do what the comment suggests:

```sh
v=$($SHELL << 'EOF'
# test code here
EOF
) && [[ "$v" == "a= b= c=0" ]] || err_exit 'variables set in subshells are not confined to the subshell'
```

Up to you which you would prefer.

Reply:
The comment and commit message have been fixed.